pydarn/plotting/grid.py

```diff
@@ -0,0 +1,261 @@
+"""
+
+import datetime as dt
+import matplotlib.pyplot as plt
+import numpy as np
+import cmocean
+
+from matplotlib import ticker, cm, colors
+from typing import List
+
+# Third party libraries
+import aacgmv2
+
+from pydarn import PyDARNColormaps, Fan, plot_exceptions
+
+
+class Grid():
+    """
+        Grid plots for SuperDARN data
+        This class inherits from matplotlib to generate the figures
+        Methods
+        -------
+        plot_grid
+    """
+    def __str__(self):
+        return "This class is static class that provides"\
+                " the following methods: \n"\
+                "   - plot_grid()\n"
+
+    @classmethod
+    def plot_grid(cls, dmap_data: List[dict], record: int = 0,
+                  start_time: dt.datetime = None, time_delta: int = 1,
+                  ax=None, fov: bool = True, parameter: str = 'vel',
+                  lowlat: int = 50, cmap: str = None, zmin: int = None,
+                  zmax: int = None, colorbar: bool = True,
+                  colorbar_label: str = '', title: str = '',
+                  len_factor: float = 150.0, ref_vector: int = 300):
+        """
+        Plots a radar's gridded vectors from a GRID file
+
+        Parameters
+        -----------
+            dmap_data: List[dict]
+                Named list of dictionaries obtained from SDarn_read
+            record: int
+                record number to plot
+                default: 0
+            start_time: datetime.datetime
+                datetime object as the start time of the record to plot
+                if none then record will be used
+                default: none
+            time_delta: int
+                How close the start_time has to be start_time of the record
+                in minutes
+                default: 1
+            ax: matplotlib.pyplot axis
+                Pre-defined axis object to pass in, must currently
+                be polar projection
+                Default: Generates a polar projection for the user
+                with MLT/latitude labels
+            parameter: str
+                Key name indicating which parameter to plot.
+                Default: vel (Velocity). Alternatives: 'pwr', 'wdt'
+            lowlat: int
+                Lower AACGM latitude boundary for the polar plot
+                Default: 50
+            fov: bool
+                Set to false to not plot the outline of the FOV
+                Default: True
+            cmap: matplotlib.cm
+                matplotlib colour map
+                https://matplotlib.org/tutorials/colors/colormaps.html
+                Default: Official pyDARN colour map for given parameter
+            zmin: int
+                The minimum parameter value for coloring
+                Default: {'pwr': [0], 'vel': [0], 'wdt': [0]}
+            zmax: int
+                The maximum parameter value for  coloring
+                Default: {'pwr': [50], 'vel': [1000], 'wdt': [250]}
```

Comment:
Are these different defaults to other plots?

Reply:
It makes sense for the defaults to be different for `vel`, as the absolute magnitude is being plotted instead of a LOS value (which can be negative). But the `zmin` and `zmax`, for `pwr` and `wdt`, should probably be the same as the range time plots if they're not already.

Reply:
It makes sense for the defaults to be different for `vel`, as the absolute magnitude is being plotted instead of a LOS value (which can be negative). But the `zmin` and `zmax`, for `pwr` and `wdt`, should probably be the same as the range time plots if they're not already.

Reply:
Correct